src/helpers/propPathOr.js

```diff
@@ -1,31 +1,45 @@
 /** @license ISC License (c) copyright 2017 original and current authors */
 /** @author Henrique Limas */
+/** @author Ian Hofmann-Hicks */
 
 const curry = require('../core/curry')
 const isArray = require('../core/isArray')
+const isEmpty = require('../core/isEmpty')
 const isInteger = require('../core/isInteger')
-const isNil= require('../core/isNil')
+const isDefined = require('../core/isDefined')
+const isNil = require('../core/isNil')
 const isString = require('../core/isString')
 
 // propPathOr : a -> [ String | Integer ] -> b -> c
 function propPathOr(def, keys, target) {
   if(!isArray(keys)) {
-    throw new TypeError('propPathOr: Array of strings or integers required for second argument')
+    throw new TypeError(
+      'propPathOr: Array of Non-empty Strings or Integers required for second argument'
+    )
   }
 
-  const value = keys.reduce((target, key) => {
-    if (isNil(target)) {
-      return target
-    }
+  if(isNil(target)) {
```

Comment:
I think we need to put that condition inside the `for`  loop too because we could have an object like `{ foo: null }` and calling the props like `['foo', 'bar']`. This will throw an error. I think it's nice to add a test for that too. IMHO The same needs to be done on `propPath` too.

Reply:
Good find!